### ISSUE TITLE
salt always targets dead minions

### ISSUE THREAD

**Issue by angeloudy** [saltstack/salt#47414]
### Description of Issue/Question
When I run 
```
salt -I role:webserver state.apply apache
```
it matches all the minions with the pillar data role:webserver, but it also matches all the dead minions which do not contain the pillar data role:webserver.
### Setup
FreeBSD 11.1
### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
Salt Version:
           Salt: 2017.7.4

Dependency Versions:
           cffi: 1.11.5
       cherrypy: unknown
       dateutil: 2.7.2
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: 1.3.12
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.6.5 (default, Apr  5 2018, 13:37:49)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 17.0.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.3

System Versions:
           dist:
         locale: UTF-8
        machine: amd64
        release: 11.1-RELEASE-p9
         system: FreeBSD
        version: Not Installed
```

**Comment by viq** [comment]
I wasn't able yet to find the docs to prove it, but that's how salt works. It uses an event bus, and unless you're targeting a specific single minion, then it's basically "hey, everyone, those of you who match this criteria, please do this thing and report back", and then waits for the results to be returned. I don't know whether there are returns on "not me", but certainly those that are offline do not respond, therefore hitting a timeout that you're seeing.

**Comment by garethgreenaway** [comment]
This should be resolved in 2018.3 with the addition of https://github.com/saltstack/salt/pull/42915
